packages/engine/tests/experiment/mod.rs

```diff
@@ -347,6 +347,18 @@ pub async fn run_test<P: AsRef<Path>>(
             Ok((json_state, globals, analysis_outputs))
         })
         .collect::<Result<_>>()?;
+
+    match fs::read_dir(output_base_directory.join((last_sim_id + 2).to_string()))
```

Comment:
Instead of creating an iterator over the files, you can simply use [`Path::exists`](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists).